compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -1395,7 +1395,7 @@ abstract class GenJSCode extends plugins.PluginComponent
 
       val ctorToChildren = secondaryCtors.map { ctor =>
         findCtorForwarderCall(ctor.body.get) -> ctor
-      }.groupBy(_._1).mapValues(_.map(_._2)).withDefaultValue(Nil)
+      }.groupBy(_._1).mapValues(_.map(_._2)).toMap.withDefaultValue(Nil)
```

Comment:
`mapValues` explicitly return a `View` in the new collections (instead of a `Map`), hence the explicit call to `toMap` to evaluate it as a `Map`.